krabs/krabs/ut.hpp

```diff
@@ -164,7 +164,7 @@ namespace krabs { namespace details {
             std::vector<BYTE> filterEventIdBuffer;
             auto filterEventIdCount = settings.provider_filter_event_ids_.size();
 
-            if (filterEventIdCount > 0) {
+            if (filterEventIdCount > 0 && filterEventIdCount <= MAX_EVENT_FILTER_EVENT_ID_COUNT) {
```

Comment:
This should be an error condition that is bubbled up to the caller and then we can either:
* Return an error condition (throw or return an error code)
* Use the first MAX_EVENT_FILTER_EVENT_ID_COUNT items

I prefer throwing so it's obvious what went wrong.

Reply:
Both variants not ideal. We limit the user's use of krabs filtering. In my case we just "turn off" system prefiltration. We will get all events from provider but have a chance to filter them from in callback.